src/main/scala/pbdirect/PBReader.scala

```diff
@@ -159,6 +167,14 @@ trait PBParserImplicits extends LowPriorityPBParserImplicits {
     instance { (index: Int, bytes: Array[Byte]) =>
       parser.parse(index, bytes).toMap
     }
+  implicit def collectionMapParser[K, V](implicit parser: PBParser[List[(K, V)]]): PBParser[collection.Map[K, V]] =
```

Comment:
what's the need for this? Isn't the `Map` parser good enough if you `import scala.collection.Map`?

Reply:
In predef.scala:  `type Map[A, +B] = immutable.Map[A, B]`

And, `immutable.Map` extends `scala.collection.Map`.

So the `mapWriter` and `mapParser` does not cover the type `scala.collection.Map`

You can try removing this parser and you'll find the test does not compile.

Unfortunately, `collection.Map` is being used (and required) by some use cases of Spark and can't just be replaced with `Map`.

Reply:
ok that makes sense